ipatests/test_integration/test_installation.py

```diff
@@ -1108,6 +1115,79 @@ def run_request(url, expected_stdout=None, expected_stderr=None):
                 expected_stdout=f'href="https://{self.master.hostname}/'
             )
 
+    @master_unistalled
+    def test_hostname_parameter(self):
+        """
+        Test for #2692 ipa-server-install ignores --hostname
+
+        https://pagure.io/freeipa/issue/2692
+        """
+        original_hostname = self.master.hostname
+        new_hostname = "new." + original_hostname
+        ip_address = self.master.ip
+        hosts = self.master.get_file_contents('/etc/hosts')
+        new_hosts = str(hosts) + "\n{}        {}".format(
+            ip_address, new_hostname)
+        self.master.put_file_contents('/etc/hosts', new_hosts)
+        result = tasks.install_master(self.master,
+                                      setup_dns=False,
+                                      extra_args=['--hostname',
+                                                  new_hostname])
+        try:
+            assert result.returncode == 0
+            hostname = self.master.run_command([
+                'hostname', '-f']).stdout_text.strip()
+            assert hostname == new_hostname
+        finally:
+            self.master.hostname = original_hostname
+            self.master.put_file_contents('/etc/hosts', hosts)
+
+    @master_unistalled
+    def test_ad_subpackage_dependency(self):
+        """
+        Test for #4011 ipa-server-install crashes
+        when AD subpackage is not installed
+
+        https://pagure.io/freeipa/issue/4011
+        """
+        package_name = "*ipa-server-trust-ad"
+        tasks.uninstall_packages(self.master, [package_name])
+        result = tasks.install_master(self.master)
+        assert result.returncode == 0
+
+    @master_unistalled
+    def test_backup_of_cs_cfg_should_be_created(self):
+        """
+        Test for #4166 Backup CS.cfg before modifying it
+
+        https://pagure.io/freeipa/issue/4166
+        """
+        bcp_location = paths.CA_CS_CFG_PATH + ".ipabkp"
+        tasks.install_master(self.master)
+        ipaserver_install_log = str(self.master.get_file_contents(
+            paths.IPASERVER_INSTALL_LOG))
+        assert "backing up CS.cfg" in ipaserver_install_log
+        try:
+            self.master.get_file_contents(bcp_location)
+        except IOError:
+            raise AssertionError("Backup of CS.cfg not created")
+
+    @master_unistalled
+    def test_update_krb5_conf_template_with_keyring_default_ccache_name(self):
+        """
+        Test for #4013 Update krb5.conf.template to support
+        KEYRING default_ccache_name
+
+        https://pagure.io/freeipa/issue/4013
+        """
+        tasks.uninstall_master(self.master)
+        tasks.install_master(self.master)
+        tasks.kinit_admin(self.master)
+        klist = self.master.run_command(['klist']).stdout_text
+        assert '/tmp/' not in klist
+        krb5conf = str(self.master.get_file_contents('/etc/krb5.conf'))
+        assert "default_ccache_name = KEYRING:persistent:%{uid}" in krb5conf
```

Comment:
This line can be present in file but commented out. We should check if it is really in effect